From ab69e4fc976d02298424f142f0a61de671a2a21c Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Fri, 5 Sep 2014 04:09:47 +0100
Subject: [PATCH 091/120] action: fix move argument parsing

At one point in time it was anticipated that the move action
might take multiple arguments.  So it used the multi argument
setting of "-2"...

But the current implementation expects one argument only.  But
the -2 setting allows zero arguments which isn't expected.  So
enforce 1 argument.

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/action.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/squashfs-tools/action.c b/squashfs-tools/action.c
index 170d095..2f2b4b5 100644
--- a/squashfs-tools/action.c
+++ b/squashfs-tools/action.c
@@ -3148,7 +3148,7 @@ static struct action_entry action_table[] = {
 	{ "guid", GUID_ACTION, 2, ACTION_ALL_LNK, parse_guid_args, guid_action},
 	{ "mode", MODE_ACTION, -2, ACTION_ALL, parse_mode_args, mode_action },
 	{ "empty", EMPTY_ACTION, -2, ACTION_DIR, parse_empty_args, NULL},
-	{ "move", MOVE_ACTION, -2, ACTION_ALL_LNK, NULL, NULL},
+	{ "move", MOVE_ACTION, 1, ACTION_ALL_LNK, NULL, NULL},
 	{ "prune", PRUNE_ACTION, 0, ACTION_ALL_LNK, NULL, NULL},
 	{ "chmod", MODE_ACTION, -2, ACTION_ALL, parse_mode_args, mode_action },
 	{ "", 0, -1, 0, NULL, NULL}
-- 
2.8.3

